### ISSUE TITLE
ERROR TypeError: this._agAwareComponent.agInit is not a function

### ISSUE THREAD

**Issue by Rajan-corient** [ag-grid/ag-grid-angular#274]
Getting error ERROR TypeError: this._agAwareComponent.agInit is not a function in angular-ag-grid.

![ag-grid error](https://user-images.githubusercontent.com/31067215/62526091-35c03d00-b856-11e9-9bed-5c172267108e.PNG)

here is my html  and ts  code

<input type="file" style="display: inline-block;" (change)="incomingfile($event)" placeholder="Upload file" accept=".xlsx">
<button type="button" class="btn btn-info" (click)="Upload()" >Upload</button>

<ag-grid-angular #agGrid
    style="height: 1000px;" 
    class="ag-theme-balham"
    [rowData]="rowData" 
    [columnDefs]="columnDefs"
    >
</ag-grid-angular>

  rowData:any;
  arrayBuffer:any;
  file:File;

Upload() {
        let fileReader = new FileReader();
          fileReader.onload = (e) => {
              this.arrayBuffer = fileReader.result;
              var data = new Uint8Array(this.arrayBuffer);
              var arr = new Array();
              for(var i = 0; i != data.length; ++i) arr[i] = String.fromCharCode(data[i]);
              var bstr = arr.join("");
              var workbook = XLSX.read(bstr, {type: 'binary', cellDates:true, cellStyles:true});
              console.log(workbook);
              var first_sheet_name = workbook.SheetNames[0];
              var worksheet = workbook.Sheets[first_sheet_name];
              console.log(XLSX.utils.sheet_to_json(worksheet,{raw:true}));
              this.rowData = XLSX.utils.sheet_to_json(worksheet,{raw:true});
              console.log(11, this.rowData);
          }
          fileReader.readAsArrayBuffer(this.file);
  }
 incomingfile(event){
    this.file= event.target.files[0]; 
    console.log(event)
  }

columnDefs:any = [
    {headerName: 'Name', field: 'name', cellRendererFramework: AppComponent, editable:true},
    {headerName: 'Company', field: 'company'},
    {headerName: 'Title', field: 'title'},
    {headerName: 'Age', field: 'age'},
    {headerName: 'Country', field: 'country'},
    {headerName: 'Salary', field: 'salary'},
  ];

**Comment by dmiko1994** [comment]
Kind of hard to follow your code here, but I had the same issue and found literally nothing on the issue except for this unanswered post. I was creating a custom filter. My issue was that I was typing params in agInit as IFilterParams as denoted in ag-grid documentation:

```
agInit(params: IFilterParams): void {
        this.params = params;
        this.valueGetter = params.valueGetter;
}
```

If I type my params as ```params: any``` I don't get that error. Maybe something new with a new version of ag-grid? Not sure.
